src/main/java/de/uol/pgdoener/civicsage/index/SemanticSplitterService.java

```diff
@@ -0,0 +1,19 @@
+package de.uol.pgdoener.civicsage.index;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import org.springframework.ai.document.Document;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class SemanticSplitterService {
+
+    public List<Document> process(@NonNull List<Document> documents) {
+        // TODO split chunks into semantically meaningful chunks
```

Comment:
When implementing the semantic splitting, we need to make sure, that the other `DocumentReader` Services provide enough information to be able to draw semantic conclusions.

Reading up on the `MarkdownDocumentReader` I can see the following: 
```
Reads the given Markdown resource and groups headers, paragraphs, or text divided by horizontal lines (depending on the MarkdownDocumentReaderConfig.horizontalRuleCreateDocument configuration) into Documents.
```

Seeing this, we might only get small chunks that may or may not have a semantic connection. So maybe we need to consider not using the default `DocumentReader`s when going the semantic chunking approach. For now I think the default implementations will be good enough though for us to keep going.

Reply:
I've just seen #8, so I think, the discussion on this topic should continue in the scope of this issue.